### PR TITLE
Add scala Eventstream sample

### PR DESCRIPTION
**Pull request by tdhd**
This is a small sample demonstrating the use of the event stream in actor systems.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @tdhd,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

**Comment by ktoso** [comment]
Oh an entirely new one. Hm, let me ask a related question – do you think the docs on the eventStream are lacking in the docs? If so, maybe they'd need more examples instead of another activator template?

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def props(ttl: FiniteDuration): Props = Props(new DyingActor(ttl))
+  case class DyingCry(message: String)
+}
+
+class DyingActor(timeToLive: FiniteDuration) extends Actor with ActorLogging {
+  import DyingActor._
+  import context.dispatcher
+
+  // scheduled to die
+  val scheduler: Cancellable = context.system.scheduler.scheduleOnce(timeToLive)(die)
```
Comment (ktoso) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala]: in general we do not encourage that `scheduleOnce` call, it is safer to use a message to `self` and in there call the die

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  def die() = {
+    context.system.eventStream.publish(DyingCry(s"${context.parent} let me die after $timeToLive!"))
+    scheduler.cancel
```
Comment (ktoso) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala]: no need to cancel a `scheduleOnce`

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import scala.concurrent.duration._
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.actor.ActorLogging
+import akka.actor.Props
+import akka.actor.Cancellable
+
+object DyingActor {
+  def props(ttl: FiniteDuration): Props = Props(new DyingActor(ttl))
+  case class DyingCry(message: String)
```
Comment (ktoso) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala]: not sure about the message name... sounds quite drastic :)

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      context.watch(context.actorOf(DyingActor.props(timeToLive), s"dyingActor$i"))
+    }
+  }
+
+  def receive = {
+    case Terminated(ref) =>
+      log.info("{} terminated, {} children left", ref, context.children.size)
+      if (context.children.size.equals(1)) {
+        // stop the listening actor
+        context.children.foreach(context.stop)
```
Comment (ktoso) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala]: that's already done in `super.postStop`, so no need to do it explicitly

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    for (i <- 1 to nActors) {
+      val timeToLive = scala.util.Random.nextFloat.seconds
+      context.watch(context.actorOf(DyingActor.props(timeToLive), s"dyingActor$i"))
+    }
+  }
+
+  def receive = {
+    case Terminated(ref) =>
+      log.info("{} terminated, {} children left", ref, context.children.size)
+      if (context.children.size.equals(1)) {
```
Comment (ktoso) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala]: in Scala `equals` is `==`

**Comment by tdhd** [comment]
The docs are good on event streams and have many good code samples. I implemented this when I first worked with event streams and schedulers. I think it might help others see a complete and simple example.

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala (hunk 5 above)
Comment (tdhd) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/ParentActor.scala]: Thanks for checking, I fixed it.

File: akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala (hunk 3 above)
Comment (tdhd) [inline akka-samples/akka-sample-eventstream-scala/src/main/scala/sample/eventstream/DyingActor.scala]: Uh, yes :). I was thinking the same. I will change it.

**Comment by tdhd** [comment]
It's fine if you don't want to merge it. I actually think that it's not contributing a whole lot.

Thanks for the code review though. :)

**Pull request closed without merging**